File: tests/fixtures/v2-html-flow/positive/HtmlFlow.java
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.owasp.encoder.Encode;

class HtmlFlow {
    void direct(HttpServletRequest request, HttpServletResponse response) throws Exception {
        response.setContentType("text/html");
        response.getWriter().write(request.getParameter("direct"));
    }

    void propagated(HttpServletRequest request, HttpServletResponse response) throws Exception {
        response.setContentType("text/html");
        String content = request.getParameter("propagated");
        String alias = content;
        response.getWriter().write(alias);
    }

    void encoded(HttpServletRequest request, HttpServletResponse response) throws Exception {
        response.setContentType("text/html");
        String content = Encode.forHtml(request.getParameter("encoded"));
        response.getWriter().write(content);
    }
}

File: tests/fixtures/v2-html-flow/positive/html_flow.go
package htmlflow

func direct(r *Request) {
	_ = template.HTML(r.FormValue("direct"))
}

func propagated(r *Request) {
	content := r.FormValue("propagated")
	alias := content
	_ = template.HTML(alias)
}

func encoded(r *Request) {
	content := html.EscapeString(r.FormValue("encoded"))
	_ = template.HTML(content)
}
